server/gitrest/packages/gitrest-base/src/utils/redisFs/redisFsManager.ts

```diff
@@ -0,0 +1,457 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type {
+	BigIntStats,
+	BufferEncodingOption,
+	Dirent,
+	MakeDirectoryOptions,
+	Mode,
+	OpenMode,
+	PathLike,
+	RmDirOptions,
+	RmOptions,
+	StatOptions,
+	Stats,
+	ObjectEncodingOptions,
+} from "fs";
+import { FileHandle } from "fs/promises";
+import { Stream } from "stream";
+import { Abortable } from "events";
+import * as IoRedis from "ioredis";
+import sizeof from "object-sizeof";
+import { getRandomInt } from "@fluidframework/server-services-client";
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+import { IFileSystemManager, IFileSystemPromises } from "../definitions";
+import { getStats, ISystemError, packedRefsFileName, SystemErrors } from "../fileSystemHelper";
+import { Redis, RedisParams } from "./redis";
+
+export interface RedisFsConfig {
+	enableRedisFsMetrics: boolean;
+	redisApiMetricsSamplingPeriod: number;
+}
+
+export class RedisFsManager implements IFileSystemManager {
+	// isomorphic-git assumes promise-style APIs from the file system implementation
+	// For example, if the file system implementation given to isomorphic-git
+	// is `fs`, it would do `fs.promises.readfile()`. Therefore, we wrap `RedisFs`
+	// in `RedisFsManager`, with a `promises` property
+	public readonly promises: IFileSystemPromises;
+	constructor(
+		redisParam: RedisParams,
+		redisOptions: IoRedis.RedisOptions,
+		redisFsConfig: RedisFsConfig,
+	) {
+		this.promises = new RedisFs(redisParam, redisOptions, redisFsConfig);
+	}
+}
+
+export class RedisFs implements IFileSystemPromises {
+	public readonly redisFs: Redis;
```

Comment:
Should we rename this to `redicClient` or similar? The current class is already called `RedisFs`, so having this property be called `redisFs` makes it a bit confusing as this is actually the Redis client/SDK implementation.